packages/protocol/contracts/L1/tiers/TaikoA6TierProvider.sol

```diff
@@ -82,8 +84,11 @@ contract TaikoA6TierProvider is ITierProvider {
     }
 
     function getMinTier(uint256 rand) public pure override returns (uint16) {
-        if (rand % 1000 == 0) return LibTiers.TIER_SGX_AND_PSE_ZKEVM;
-        else if (rand % 100 == 0) return LibTiers.TIER_SGX;
+        if (rand % 500 == 0) return LibTiers.TIER_SGX_AND_PSE_ZKEVM; // 0.2% of the blocks
```

Comment:
One remaining concern here is this randomly selected block is unprovable due to unsupported tx, high rws, or some unknown issues(hidden bugs). 
For example, if a popular partner who still use legacy tx launches on taiko, like the one we met in A5, for a period of time all blocks are filled with at least on of their txs, which makes all blocks in that period unprovable, no matter what the rand is.
Should we have provable blocks (could be pure anchor block) at fixed intervals playing as ZK checkpoints without purely rely on rand?

Reply:
How can you have provable blocks at fixed intervals? If a block is not provable, we need to prove it using our guardian prover.

Reply:
Yes, here we rely heavily on oracle, I just thought besides the rand picking, maybe we can intentionally submit a few zk blocks to kind of refresh the zk ratio requirement.

Reply:
We can certainly do that, and prove it proactively even with ZKPs even if ZKPs are not required.